lib/perl/Genome/Qc/Tool/Picard.pm

```diff
@@ -0,0 +1,53 @@
+package Genome::Qc::Tool::Picard;
+
+use strict;
+use warnings;
+use Genome;
+
+class Genome::Qc::Tool::Picard {
+    is => 'Genome::Qc::Tool',
+    is_abstract => 1,
+};
+
+sub cmd_line {
+    my $self = shift;
+    my $cmd = $self->gmt_class->create($self->gmt_params);
+    return $cmd->build_cmdline_list;
+}
+
+sub get_metrics {
+    my $self = shift;
+
+    my $file = $self->output_file;
+    my $gmt_class = $self->gmt_class;
+    my %metrics = $self->metrics;
+    my %metric_results = %{$gmt_class->parse_file_into_metrics_hashref($file)};
+    return $self->_get_metrics(\%metrics, \%metric_results);
+}
+
+sub metrics {
+    my $self = shift;
+    die $self->error_message("Abstract method metrics must be overridden by subclass");
+}
+
+sub gmt_class {
+    my $self = shift;
+    die $self->error_message("Abstract method gmt_class must be overridden by subclass");
+}
+
+sub _get_metrics {
+    my ($self, $metrics, $metric_results) = @_;
+
+    my %desired_metric_results;
+    while (my ($metric, $metric_details) = each %{$metrics}) {
+        my $metric_key = $metric_details->{metric_key};
+        unless (defined($metric_key)) {
+            ($metric_key) = keys %{$metric_results};
```

Comment:
If the Picard gmt class doesn't have _metric_header_as_key defined than the metric_key during creation of the hash will be set to the first column's header and value. As far as I can tell this is the case when the metrics are output on only one line and aren't subcategorized. When _metric_header_as_key is defined this subroutine defines the column that is used for subcategorizing (e.g., read1, read2, both) where each key then contains values that are applicable to that category (e.g., % reads aligned for the read1 category). In the rest of the cases we don't have categories so the metrics_key is just a useless middle layer.